app/controllers/application_controller.rb

```diff
@@ -2,4 +2,59 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :current_user
+  before_action :current_order
+  before_action :items_in_cart
+
+
+  def current_order
+    if session[:order_id] && !!Order.exists?(session[:order_id])
+    @current_order ||= Order.find(session[:order_id])
+    end
+  end
+
+  def current_user
+    if session[:user_id] && !!User.exists?(session[:user_id])
+      @current_user ||= User.find(session[:user_id])
+    end
+  end
+
+  def items_in_cart
+    if current_order
+      sum = []
+      Order.find(session[:order_id]).orderitems.each do |oi|
+        sum.push(oi.quantity)
+      end
+      @cart_num = sum.inject(0) {|r, e| r + e }
```

Comment:
Try to lean more towards descriptive variables vs. letters. It makes your code much easier to read.